### PR TITLE
Draft Implementation of time integration class

### PR DESCRIPTION
**Pull request by hqwang815**

### CODE REVIEW

**Comment by hqwang815** [comment]
Initial completion of the DG code.

**Review by apalha — CHANGES_REQUESTED** [review]
@hqwang815 , I finished reviewing and adding my comments. I propose you go over them and then we arrange a meeting and discuss this.

File: edg_acoustics/acoustics_simulation.py [hunk 1]
```diff
@@ -125,17 +126,20 @@ class AcousticsSimulation:
     """
 
   
-    def __init__(self, Nx: int, Nt: int, mesh: edg_acoustics.Mesh, BC_list: dict[str, int], node_tolerance: float = NODETOL):
+    def __init__(self, rho0: float, c0: float, Nx: int, Nt: int, mesh: edg_acoustics.Mesh, BC_list: dict[str, int], node_tolerance: float = NODETOL):
```
[the remark this answers is not in the document]
Comment (hqwang815) [reply edg_acoustics/acoustics_simulation.py:129]: Yes, I was wondering the same before. Now, I added as you suggested

File: edg_acoustics/acoustics_simulation.py [hunk 2]
```diff
@@ -876,8 +857,231 @@ def build_BCmaps_3d(BC_list: dict[str, int], EToV: numpy.ndarray, vmapM: numpy.n
...
+    @staticmethod
+    def factorial(n):
+        if n == 0 or n == 1:
+            return 1
+        else:
+            return n * AcousticsSimulation.factorial(n - 1)
+        
+    @staticmethod
+    #https://stackoverflow.com/questions/25179693/how-to-check-whether-the-point-is-in-the-tetrahedron-or-not/60745339#60745339
+    def locate_simplex(node_coordinates: numpy.ndarray, node_ids: numpy.ndarray, rec: numpy.ndarray):
```
[the remark this answers is not in the document]
Comment (hqwang815) [reply edg_acoustics/acoustics_simulation.py:899]: Thanks. I write it as a static method and explicitly pass `node coordinates`, `node_ids` in order to make it possible to call this function from another code, for example my colleague's code that needs the same function. Of course, we can make our own user's life easier. I will change it as you suggest and add the options.

File: edg_acoustics/acoustics_simulation.py [hunk 3]
```diff
@@ -876,8 +857,231 @@ def build_BCmaps_3d(BC_list: dict[str, int], EToV: numpy.ndarray, vmapM: numpy.n
...
+
+        
+        Args:
+
+        Returns:
+        """
+        self.Flux = Flux
+        # self.Flux = edg_acoustics.UpwindFlux(self.rho0, self.c0, self.n_xyz)
+
+    def init_TimeIntegration(self, TI: edg_acoustics.TimeIntegrator, rec: numpy.ndarray, TotalTime: float=0.05):
```
[the remark this answers is not in the document]
Comment (hqwang815) [reply edg_acoustics/acoustics_simulation.py:1015]: Yes, i implement `init_rec` inside AcousticSimulation class and it takes `rec` as input.
In the future, we might need to add functionality to read input of text file later. I add this functionality in the issue to be solved. Let's just keep it simple for now.

File: edg_acoustics/acoustics_simulation.py [hunk 4]
```diff
@@ -876,8 +857,231 @@ def build_BCmaps_3d(BC_list: dict[str, int], EToV: numpy.ndarray, vmapM: numpy.n
...
+
+        Returns:
+            sampleWeight (numpy.ndarray): ``[N_rec, Np]`` interpolation weights required to interpolate the nodal data to the sample (i.e., microphone location).
+            nodeindex (numpy.ndarray): ``[N_rec, ]`` index of simplices that contatin the sample (microphone) points
+        """
+        tri = Delaunay(vertices.T, qhull_options = 'QJ')
+        tri.simplices = EToV.T
+        tri.nsimplex = EToV.shape[1]
+
+        nodeindex = tri.find_simplex(rec.T)
```
[the remark this answers is not in the document]
Comment (hqwang815) [reply edg_acoustics/acoustics_simulation.py:963]: Yes, like I reply above. I will add two options.

**Pull request merged**